whylog/rulesbase/__init__.py

```diff
@@ -1,6 +1,144 @@
+import re
+import yaml
+from abc import ABCMeta, abstractmethod
+from collections import defaultdict
 
+class AbstractConfigAccess(object):
+    __metaclass__ = ABCMeta
 
-class WhylogBase(object):
+    @abstractmethod
+    def get_parsers(self):
+        pass
 
-    def __init__(self, **kwargs):
+    @abstractmethod
+    def get_rule(self, effect_name):
         pass
+        
+class YamlConfigAccess(AbstractConfigAccess):
```

Comment:
contents of most methods of this class are not specific to yaml format. There should be an intermediate class or two here, like `FileBasedConfig` or something